tests/jballerina-unit-test/src/test/resources/test-src/expressions/access/field_access.bal

```diff
@@ -353,10 +353,45 @@ function testFieldAccessOnMapConstruct() returns boolean {
     return "Sanjiva" == name;
 }
 
+class EmployeeR {
+    function func(int i) returns int => i;
+}
+
+class ManagerR {
+    function func(int i) returns int => i + 1;
+}
+
+class CompanyR {
+    function func(int i) returns string => (i + 2).toString();
+}
+
+function testFieldAccessOnUnion() {
+    EmployeeR|ManagerR ob1 = new ManagerR();
+    function (int i) returns int func1 = ob1.func;
+    assertEquals(func1(1), 2);
+
+    ManagerR ob2 = new EmployeeR();
+    EmployeeR|ManagerR ob3 = ob2;
+    function (int i) returns int func2 = ob3.func;
+    assertEquals(func2(1), 1);
+
+    CompanyR|ManagerR ob4 = new ManagerR();
+    (function (int i) returns string)|function (int i) returns int func3 = ob4.func;
+    function (int i) returns int func4 = <function (int i) returns int> func3;
+    assertEquals(func4(1), 2);
+}
+
```

Comment:
```ballerina
class EmployeeR {
    function func(int i) returns int => i;
}

class ManagerR {
    function func(int i) returns int => i + 1;
}


function dummyTest() {
    EmployeeR|ManagerR p = new ManagerR();
    print(p.func(1)); // @error undefined function 'func' in type '(EmployeeR|ManagerR)']
}

```

Here, the error is not correct because both members have the same func

Reply:
Is this error acceptable here `function invocation on type '(function (int) returns (int)|function (int) returns (int))' is not supported` ?

Reply:
According to the conclusion in this issue https://github.com/ballerina-platform/ballerina-lang/issues/19061, the error msg seems correct.